src/core/deltafilewrapper.cpp

```diff
@@ -892,17 +778,187 @@ void DeltaFileWrapper::addCreate( const QString &localLayerId, const QString &so
 
   delta.insert( QStringLiteral( "new" ), newData );
 
-  QString localPk = delta.value( QStringLiteral( "localPk" ) ).toString();
-  mLocalPkDeltaIdx[localLayerId][localPk] = static_cast<int>( mDeltas.count() );
+  appendDelta( delta );
+}
 
-  mDeltas.append( delta );
-  mIsDirty = true;
+void DeltaFileWrapper::appendDelta( const QJsonObject &delta )
+{
+  mergeDelta( delta );
+}
 
-  qInfo() << "DeltaFileWrapper::addCreate: Added a new create delta: " << delta;
+void DeltaFileWrapper::mergeDelta( const QJsonObject &delta )
```

Comment:
Can we break up this method into three, instead of three gigantic `if` bodies:

```
if (deltamethod == "create" ) {
   appendCreateDelta( delta );
} else if (deltamethod == "patch" ) {
   appendPatchDelta( delta );
} else if (deltamethod == "delete" ) {
   appendDeleteDelta( delta );
} else {
   Q_ASSERT( 0 );
}
```

We can save a bit of nesting and improve readability.

Reply:
@suricactus , we can, but maybe just adding some // Comment with more spacing to identify the sections would be OK? I'm indifferent really :)